src/eos-metrics-instrumentation.c

```diff
@@ -78,6 +90,96 @@ static gint64 start_time;
 
 static EinsPersistentTally *persistent_tally;
 
+static gboolean
+get_os_version (char **name_out,
```

Comment:
The `gchar` types are considered legacy (and perhaps deprecated) upstream, which is why I don't use them.